okio/native/src/main/kotlin/okio/ByteString.kt

```diff
@@ -0,0 +1,214 @@
+  internal actual open fun getByte(pos: Int) = commonGetByte(pos)
+
+  /** Returns the byte at `index`.  */
+  actual operator fun get(index: Int): Byte = getByte(index)
+
+  /** Returns the number of bytes in this ByteString. */
+  actual val size
+    get() = getSize()
+
+  // Hack to work around Kotlin's limitation for using JvmName on open/override vals/funs
+  internal actual open fun getSize() = commonGetSize()
+
+  /** Returns a byte array containing a copy of the bytes in this `ByteString`. */
+  actual open fun toByteArray() = commonToByteArray()
+
+  /** Returns the bytes of this string without a defensive copy. Do not mutate!  */
+  internal actual open fun internalArray() = commonInternalArray()
+
+  /**
+   * Returns true if the bytes of this in `[offset..offset+byteCount)` equal the bytes of `other` in
+   * `[otherOffset..otherOffset+byteCount)`. Returns false if either range is out of bounds.
+   */
+  actual open fun rangeEquals(
+    offset: Int,
+    other: ByteString,
+    otherOffset: Int,
+    byteCount: Int
+  ): Boolean = commonRangeEquals(offset, other, otherOffset, byteCount)
+
+  /**
+   * Returns true if the bytes of this in `[offset..offset+byteCount)` equal the bytes of `other` in
+   * `[otherOffset..otherOffset+byteCount)`. Returns false if either range is out of bounds.
+   */
+  actual open fun rangeEquals(
+    offset: Int,
+    other: ByteArray,
+    otherOffset: Int,
+    byteCount: Int
+  ): Boolean = commonRangeEquals(offset, other, otherOffset, byteCount)
+
+  actual fun startsWith(prefix: ByteString) = commonStartsWith(prefix)
+
+  actual fun startsWith(prefix: ByteArray) = commonStartsWith(prefix)
+
+  actual fun endsWith(suffix: ByteString) = commonEndsWith(suffix)
+
+  actual fun endsWith(suffix: ByteArray) = commonEndsWith(suffix)
+
+  actual fun indexOf(other: ByteString, fromIndex: Int) = indexOf(other.internalArray(), fromIndex)
+
+  actual open fun indexOf(other: ByteArray, fromIndex: Int) = commonIndexOf(other, fromIndex)
+
+  // TODO move lastIndexOf() when https://youtrack.jetbrains.com/issue/KT-22818 is fixed
+
+  fun lastIndexOf(other: ByteString, fromIndex: Int = size) = lastIndexOf(other.internalArray(),
+    fromIndex)
+
+  open fun lastIndexOf(other: ByteArray, fromIndex: Int = size): Int {
+    var fromIndex = fromIndex
+    fromIndex = minOf(fromIndex, data.size - other.size)
+    for (i in fromIndex downTo 0) {
+      if (arrayRangeEquals(data, i, other, 0, other.size)) {
+        return i
+      }
+    }
+    return -1
+  }
+
+  actual override fun equals(other: Any?) = commonEquals(other)
+
+  actual override fun hashCode() = commonHashCode()
+
+  actual override fun compareTo(other: ByteString) = commonCompareTo(other)
+
+  /**
+   * Returns a human-readable string that describes the contents of this byte string. Typically this
+   * is a string like `[text=Hello]` or `[hex=0000ffff]`.
+   */
+  actual override fun toString(): String = data.toUtf8String()
```

Comment:
inconsistent with the JVM?

Reply:
I just copied what was in the Kotlin/JS module. Should I fix both?

Reply:
yes please!

Reply:
This required a lot more work than I was hoping. I've add this change as a separate commit in case you would rather not include it with this pull-request. I'm not completely pleased with how I achieved the functionality so please be very picky with your review and suggest any changes you think would be appropriate.